### DISCUSSION TITLE
Unable to do Git clone in the AWS EC2 ubuntu server

### DISCUSSION THREAD

**Discussion by Ashokkumar047** [community/community#101536] [Programming Help] [answered]
### Body

i am new to GitHub..
as i am doing first time git clone in AWS EC2 ubuntu server  via SSH using private key 
it is asking username and password while cloning if i enter proper username and password it is showing authentication failed for the my GitHub Repo 
even after try with the Personal access tokens (classic) it is showing  No such file or directory
do suggest a better way to push the data from my local or from GitHub to AWS EC2 ubuntu server
as i already installed the git in my AWS EC2 ubuntu server with the latest version

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by fury-05** [comment]
If you're facing authentication issues while cloning a GitHub repository to your AWS EC2 instance via SSH, there are a few things you can check and consider for a smoother workflow.

Make sure you have generated an SSH key pair on your local machine. You can use the following command to generate one if you haven't already:

```bash
ssh-keygen -t rsa -b 4096 -C "your_email@example.com"

```
Ensure that the public key (~/.ssh/id_rsa.pub) is added to your GitHub account. You can copy the contents of this file and add it to your GitHub account in the SSH and GPG keys settings.

I guess as your keys are not in github account you are getting failed for github repo.

**Reply by Ashokkumar047** [reply]
@fury-05 
Thanks for your quick support 
now i added the SSH public key into my GitHub org after that getting below error 

**ec2-user@1XX.X.X.X: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.**
 may i get where exactly it is missed

**Reply by fury-05** [reply, marked as answer]
Check your SSH configuration on your EC2 instance. Ensure that your SSH key is correctly configured in the ~/.ssh/config file. It should look something like this:

```bash
Host github.com
    HostName github.com
    User git
    IdentityFile ~/.ssh/id_rsa
```
Make sure the permissions on the ~/.ssh directory and the ~/.ssh/id_rsa file are secure:
```bash
chmod 700 ~/.ssh
chmod 600 ~/.ssh/id_rsa
```

Other option :

Personal Access Token: If you're still having issues, you can use a Personal Access Token instead of your GitHub password. Generate a token in your GitHub account settings, and then use it in place of your password when prompted.

Git Configuration: Double-check your global git configurations on the EC2 instance:
```yaml
git config --global user.name "Your Name"
git config --global user.email "your_email@example.com"
```
Cloning using HTTPS:

If the SSH setup continues to cause issues, consider cloning using HTTPS instead. Update your remote URL using the HTTPS URL of your repository:
```yaml
git remote set-url origin https://username@github.com/username/repo.git
```
Remember to replace "username" and "repo" with your actual GitHub username and repository name.

**Reply by Ashokkumar047** [reply]
Thanks @fury-05...
